play.pokemonshowdown.com/src/battle-team-editor.tsx

```diff
@@ -1350,6 +1353,14 @@ class TeamWizard extends preact.Component<{
 			switch (type) {
 			case 'pokemon':
 				set.species = name;
+				const species = editor.dex.species.get(name);
+				if (!(editor.format.includes('hackmons') || editor.format.endsWith('bh')) && species.requiredItems.length === 1) {
+					set.item = species.requiredItems[0];
+				} else {
+					set.item = '';
+				}
```

Comment:
```suggestion
				if (!(editor.format.includes('hackmons') || editor.format.endsWith('bh')) && species.requiredItems.length === 1) {
					set.item = species.requiredItems[0];
				}
```
Probably we don't want to clear the item if we're changing a set and the species doesn't have a required item.